pkg/server/span_download.go

```diff
@@ -89,6 +99,33 @@ func (s *systemStatusServer) DownloadSpan(
 	return resp, nil
 }
 
+func maybeFilterSpans(
+	ctx context.Context, spans []roachpb.Span, settings *cluster.Settings, nodeID roachpb.NodeID,
+) []roachpb.Span {
+	skipFraction := testingSkipSpanDownload.Get(&settings.SV)
+	if skipFraction <= 0 {
+		return spans
+	}
+
+	nodeIDBytes := []byte(nodeID.String())
+	hash := fnv.New64a()
+
+	result := []roachpb.Span{}
+	for _, span := range spans {
+		hash.Reset()
+		_, _ = hash.Write(nodeIDBytes)
```

Comment:
nit: add a comment that motivates the use of a hash here? i.e. on repeated DownloadSpan calls we want the same spans to be skipped.